server/executor/trace_poller.go

```diff
@@ -108,11 +111,38 @@ func (tp tracePoller) Stop() {
 
 func (tp tracePoller) Poll(ctx context.Context, test model.Test, run model.Run) {
 	log.Printf("[TracePoller] Test %s Run %d: Poll\n", test.ID, run.ID)
-	tp.enqueueJob(PollingRequest{
+
+	job := PollingRequest{
 		ctx:  ctx,
 		test: test,
 		run:  run,
-	})
+	}
+
+	log.Printf("[PollerExecutor] Test %s Run %d: Is Data Store Configured? %s \n", job.test.ID, run.ID, strconv.FormatBool(tp.isDataStoreConfigured))
+
+	if tp.isDataStoreConfigured {
+		tp.enqueueJob(job)
+	} else {
+		tp.syncProcessJob(job)
+	}
+}
+
+func (tp tracePoller) syncProcessJob(job PollingRequest) {
+	run := job.run
```

Comment:
this is not a great go pattern. I'd remove this and directly access the values with the extra `run.`